shopping-cart/shopping-cart-java/shopping-cart/src/main/java/com/example/shoppingcart/impl/ShoppingCart.java

```diff
@@ -0,0 +1,381 @@
+        }
+    }
+
+    @Value
+    @JsonDeserialize
+    static final class Rejected implements Confirmation {
+        public final String reason;
+
+        @JsonCreator
+        public Rejected(String reason) {
+            this.reason = reason;
+        }
+    }
+
+    //
+    // SHOPPING CART EVENTS
+    //
+    public interface Event extends Jsonable, AggregateEvent<Event> {
+        /**
+         * The tag for shopping cart events, used for consuming the Journal event stream later.
+         */
+        AggregateEventShards<Event> TAG = AggregateEventTag.sharded(Event.class, 10);
+
+        @Override
+        default AggregateEventTagger<Event> aggregateTag() {
+            return TAG;
+        }
+    }
+
+    @Value
+    @JsonDeserialize
+    static final class ItemAdded implements Event {
+        public final String shoppingCartId;
+        public final String itemId;
+        public final int quantity;
+        public final Instant eventTime;
+
+        @JsonCreator
+        ItemAdded(String shoppingCartId, String itemId, int quantity, Instant eventTime) {
+            this.shoppingCartId = Preconditions.checkNotNull(shoppingCartId, "shoppingCartId");
+            this.itemId = Preconditions.checkNotNull(itemId, "itemId");
+            this.quantity = quantity;
+            this.eventTime = eventTime;
+        }
+    }
+
+    @Value
+    @JsonDeserialize
+    static final class ItemRemoved implements Event {
+        public final String shoppingCartId;
+        public final String itemId;
+        public final Instant eventTime;
+
+        @JsonCreator
+        ItemRemoved(String shoppingCartId, String itemId, Instant eventTime) {
+            this.shoppingCartId = Preconditions.checkNotNull(shoppingCartId, "shoppingCartId");
+            this.itemId = Preconditions.checkNotNull(itemId, "itemId");
+            this.eventTime = eventTime;
+        }
+    }
+
+    @Value
+    @JsonDeserialize
+    static final class ItemQuantityAdjusted implements Event {
+        public final String shoppingCartId;
+        public final String itemId;
+        public final int quantity;
+        public final Instant eventTime;
+
+        @JsonCreator
+        ItemQuantityAdjusted(String shoppingCartId, String itemId, int quantity, Instant eventTime) {
+            this.shoppingCartId = Preconditions.checkNotNull(shoppingCartId, "shoppingCartId");
+            this.itemId = Preconditions.checkNotNull(itemId, "itemId");
+            this.quantity = quantity;
+            this.eventTime = eventTime;
+        }
+    }
+
+    @Value
+    @JsonDeserialize
+    static final class CheckedOut implements Event {
+
+        public final String shoppingCartId;
+        public final Instant eventTime;
+
+        @JsonCreator
+        CheckedOut(String shoppingCartId, Instant eventTime) {
+            this.shoppingCartId = Preconditions.checkNotNull(shoppingCartId, "shoppingCartId");
+            this.eventTime = eventTime;
+        }
+    }
```

Comment:
I know the style guide prescribes making all commands and events inner classes of the Aggregate but I think in Java this causes the Aggregate file to be bloated. In scala, thanks to the conciseness of case classes using inner classes makes code readable enough but not in Java.

Reply:
I know this comment will be controversial and I don't want to get on a ping-pong game over this PR. Feel free to ignore this.